core/evalir.ml

```diff
@@ -673,29 +673,34 @@ struct
          | Some (limit, offset) ->
             Some (Value.unbox_int (value env limit), Value.unbox_int (value env offset)) in
        if Settings.get_value Basicsettings.Shredding.shredding then
-         begin
-           match EvalNestedQuery.compile_shredded env (range, e) with
-           | None -> computation env cont e
-           | Some (db, p) ->
-             begin
-               if db#driver_name() <> "postgresql"
-                 then raise (Errors.Runtime_error "Only PostgreSQL database driver supports shredding");
-               let get_fields t =
-                             match t with
-                             | `Record fields ->
-                                 StringMap.to_list (fun name p -> (name, `Primitive p)) fields
-                             | _ -> assert false
-               in
-               let execute_shredded_raw (q, t) =
-                 Database.execute_select_result (get_fields t) q db, t in
-               let raw_results =
-                 EvalNestedQuery.Shred.pmap execute_shredded_raw p in
-               let mapped_results =
-                 EvalNestedQuery.Shred.pmap EvalNestedQuery.Stitch.build_stitch_map raw_results in
-                 apply_cont cont env
-                 (EvalNestedQuery.Stitch.stitch_mapped_query mapped_results)
-             end
-         end
+         match EvalNestedQuery.compile_shredded env (range, e) with
+         | None -> computation env cont e
+         | Some (db, p) ->
+            try
+              let driver = db#driver_name () in
```

Comment:
I'd like if we could push the knowledge of whether a database supports shredding into the db object, so that when/if we add other database drivers (or support shredding for mysql) we don't have to keep changing this line of code to have more hard-wired constants.